implementations/rs/src/lib.rs

```diff
@@ -0,0 +1,204 @@
+            }
+        } else if let Some(parameters) = params {
+            match method {
+                "eth_getBlockByNumber" => handle_get_block_by_number_params(parameters),
+                "eth_feeHistory" => vec![],
+                "eth_signTypedData_v4" => handle_sign_typed_data_args(parameters),
+                _ => from_str(parameters).unwrap(),
+            }
+        } else {
+            vec![]
+        }
+    }
+
+    fn is_transaction_method(&self, method: &str) -> bool {
+        let transaction_methods = ["eth_sendTransaction", "eth_estimateGas", "eth_call"];
+        transaction_methods.contains(&method)
+    }
+}
+
+#[plugin_impl]
+impl Module for EthereumWalletPlugin {
+    fn request(&mut self, args: &ArgsRequest, _: Arc<dyn Invoker>) -> Result<String, PluginError> {
+        let connection = self.connections.get_connection(args.connection.clone());
+        let provider: &Provider<Http> = &connection.provider;
+        let method = args.method.as_str();
+        let parameters = self.parse_parameters(method, &args.params);
+        let runtime = tokio::runtime::Runtime::new().unwrap();
+        match method {
+            "eth_signTypedData_v4" => {
+                let signer = connection.get_signer().unwrap();
+                let typed_data: TypedData = from_value(parameters[1].clone()).unwrap();
+                let hash = Runtime::block_on(&runtime, signer.sign_typed_data(&typed_data));
+                Ok(hash.unwrap().to_string())
+            }
+            "eth_sendTransaction" => {
+                let signer = connection.get_signer().unwrap();
+                let client = SignerMiddleware::new(provider, signer);
+                let tx: TransactionRequest = from_value(parameters[0].clone()).unwrap();
+                let hash = Runtime::block_on(&runtime, client.send_transaction(tx, None));
+                return Ok(hash.unwrap().to_string());
+            }
+            _ => {
+                let response = Runtime::block_on(
+                    &runtime,
+                    provider.request::<Vec<Value>, Value>(method, parameters),
+                );
+
+                let result = response.map_err(|e| e.to_string()).unwrap();
+
+                match result {
+                    Value::String(r) => Ok(r),
+                    Value::Object(object) => to_string(&object).map_err(PluginError::JSONError),
+                    _ => Ok("".to_string()),
+                }
+            }
+        }
+    }
+
+    fn wait_for_transaction(
+        &mut self,
+        args: &ArgsWaitForTransaction,
+        _: Arc<dyn Invoker>,
+    ) -> Result<bool, PluginError> {
+        let connection = self.connections.get_connection(args.connection.clone());
+        let pending_transaction = PendingTransaction::new(
+            TxHash::from_str(&args.tx_hash).unwrap(),
+            &connection.provider,
+        );
+
+        // pending_transaction.confirmations(args.confirmations.try_into().unwrap());
```

Comment:
Is this functionality not supported yet? Why is this commented out?

Reply:
i had this error:
![image](https://github.com/polywrap/ethereum-wallet/assets/21031138/00c76676-ead5-4bbe-97d3-db79d9d2b73b)
and I just did not want to debug this ownership error instantly :yum:, but will do in the coming days